### ISSUE TITLE
Publish maven artifact with CircleProcessor class

### ISSUE THREAD

**Issue by porunov** [elastic/elasticsearch#86607]
### Description

It would be really great if [CircleProcessor.java](https://github.com/elastic/elasticsearch/blob/0699c9351f1439e246d408fd6538deafde4087b6/x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/ingest/CircleProcessor.java) could be published in some maven artifact.
We are thinking to convert Geoshape Circle into Geoshape Polygon on application side instead of ElasticSearch Server side. The benefits of it is that it would make possible for us to choose `error_distance` dynamically depending on different conditions.
With direct code access to [CircleProcessor.java](https://github.com/elastic/elasticsearch/blob/0699c9351f1439e246d408fd6538deafde4087b6/x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/ingest/CircleProcessor.java) and [SpatialUtils.java](https://github.com/elastic/elasticsearch/blob/0699c9351f1439e246d408fd6538deafde4087b6/x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialUtils.java) it would be possible to make application side custom conversion logic.
ElasticSearch version: >= 8

**Comment by iverase** [comment]
We discussed in the team and we think this is a fair ask and we could move the CircleProcessor logic into the geo library which is provided as an artifact.

**Comment by porunov** [comment]
> We discussed in the team and we think this is a fair ask and we could move the CircleProcessor logic into the geo library which is provided as an artifact. 

Thank you! It would be really helpful.

**Comment by porunov** [comment]
@iverase I was trying to resolve this issue but have some problems with introducing lucene dependency to `geo` library.
Basically, `CircleProcessor` is using `SpatialUtils` class.
`SpatialUtils` class depends on `org.apache.lucene.util.SloppyMath` which is provided via `lucene-core`.
I tried to include `api "org.apache.lucene:lucene-core:${versions.lucene}"` into `libs/geo/build.gradle` but it didn't help. For some reason this library isn't included into the classpath after build is finished. Looks like some other building logic excluding `lucene-core` for some reason.
I created PR to who the problem here: #88088 
If you have any suggestions on how to resolve this issue, that would be super helpful.
